Error: Cannot find module 'core-js/modules/es6.object.assign'

<!--
  Please fill out each section below, otherwise your issue will be closed. This info allows Gatsby maintainers to diagnose (and fix!) your issue as quickly as possible.

  Useful Links:
  - Documentation: https://www.gatsbyjs.org/docs/
  - How to File an Issue: https://www.gatsbyjs.org/contributing/how-to-file-an-issue/

  Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
-->

## Description

Describe the issue that you're seeing.
When I first ran gatsby develop, I got the warning:

```
WARNING: We noticed you're using the `useBuiltIns` option without declaring a core-js version. Currently, we assume version 2.x when no version is passed. Since this default version will likely change in future versions of Babel, we recommend explicitly setting the core-js version you are using via the `corejs` option.

You should also be sure that the version you pass to the `corejs` option matches the version specified in your `package.json`'s `dependencies` section. If it doesn't, you need to run one of the following commands:

  npm install --save core-js@2    npm install --save core-js@3
  yarn add core-js@2              yarn add core-js@3
```
So I installed core-js@3. Then I ran gatsby develop again, and this time I got the error which exited me out of gatsby build:

```
error UNHANDLED REJECTION
Error: Cannot find module 'core-js/modules/es6.object.assign'
loader.js:649 Function.Module._resolveFilename
    internal/modules/cjs/loader.js:649:15

  - loader.js:575 Function.Module._load
    internal/modules/cjs/loader.js:575:25

  - loader.js:705 Module.require
    internal/modules/cjs/loader.js:705:19

  - helpers.js:14 require
    internal/modules/cjs/helpers.js:14:16

  - render-page.js:3 webpackUniversalModuleDefinition
    /Users/mariacam/Development/mariadcampbell/public/render-page.js:3:54

  - render-page.js:10 Object.<anonymous>
    /Users/mariacam/Development/mariadcampbell/public/render-page.js:10:3

  - loader.js:799 Module._compile
    internal/modules/cjs/loader.js:799:30

  - loader.js:810 Object.Module._extensions..js
    internal/modules/cjs/loader.js:810:10

  - loader.js:666 Module.load
    internal/modules/cjs/loader.js:666:32

  - loader.js:606 tryModuleLoad
    internal/modules/cjs/loader.js:606:12

  - loader.js:598 Function.Module._load
    internal/modules/cjs/loader.js:598:3

  - loader.js:705 Module.require
    internal/modules/cjs/loader.js:705:19

  - helpers.js:14 require
    internal/modules/cjs/helpers.js:14:16

  - worker.js:32 Promise
    [mariadcampbell]/[gatsby]/dist/utils/worker.js:32:35

  - debuggability.js:313 Promise._execute
    [mariadcampbell]/[bluebird]/js/release/debuggability.js:313:9

  - promise.js:483 Promise._resolveFromExecutor
    [mariadcampbell]/[bluebird]/js/release/promise.js:483:18


Maria-Campbells-Computer-2 ॐ  ~/development/mariadcampbell:(25m|git@master!)
11321 ±
```

### Steps to reproduce

Clear steps describing how to reproduce the issue. Please please please link to a demo project if possible, this makes your issue _much_ easier to diagnose (seriously).

### Expected result
What should happen?
gatsby develop should have resulted in the ability for me to view the project in the browser. 

### Actual result

What happened.
I am using the default starter. I had a similar issue when using the starter blog, but was not exited out when running gatsby develop. Also did not install core-js. Problems started occurring and I was exited out of the program when I installed core-js@3.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

```
gatsby info --clipboard                                             ⏎ ✹

  System:
    OS: macOS 10.14.3
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 11.12.0 - ~/.nvm/versions/node/v11.12.0/bin/node
    npm: 6.7.0 - ~/.nvm/versions/node/v11.12.0/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Firefox: 36.0.4
  npmPackages:
    gatsby: ^2.2.5 => 2.2.5
    gatsby-image: ^2.0.34 => 2.0.34
    gatsby-plugin-manifest: ^2.0.24 => 2.0.24
    gatsby-plugin-offline: ^2.0.25 => 2.0.25
    gatsby-plugin-react-helmet: ^3.0.10 => 3.0.10
    gatsby-plugin-sharp: ^2.0.30 => 2.0.30
    gatsby-source-filesystem: ^2.0.27 => 2.0.27
    gatsby-transformer-sharp: ^2.1.17 => 2.1.17
  npmGlobalPackages:
    gatsby-cli: 2.4.17

```

--- a/packages/babel-preset-gatsby/package.json
+++ b/packages/babel-preset-gatsby/package.json
@@ -6,7 +6,7 @@
     "@babel/plugin-proposal-class-properties": "^7.0.0",
     "@babel/plugin-syntax-dynamic-import": "^7.0.0",
     "@babel/plugin-transform-runtime": "^7.0.0",
-    "@babel/preset-env": "^7.0.0",
+    "@babel/preset-env": "^7.4.1",
     "@babel/preset-react": "^7.0.0",
     "babel-plugin-macros": "^2.4.2"
   },

--- a/packages/babel-preset-gatsby/src/__tests__/index.js
+++ b/packages/babel-preset-gatsby/src/__tests__/index.js
@@ -8,6 +8,7 @@ it(`Specifies proper presets and plugins for test stage`, () => {
     [
       expect.stringContaining(path.join(`@babel`, `preset-env`)),
       {
+        corejs: 2,
         loose: true,
         modules: `commonjs`,
         useBuiltIns: `usage`,
@@ -62,6 +63,7 @@ it(`Specifies proper presets and plugins for build-html stage`, () => {
     [
       expect.stringContaining(path.join(`@babel`, `preset-env`)),
       {
+        corejs: 2,
         loose: true,
         modules: false,
         useBuiltIns: `usage`,
@@ -109,6 +111,7 @@ it(`Allows to configure browser targets`, () => {
   expect(presets[0]).toEqual([
     expect.stringContaining(path.join(`@babel`, `preset-env`)),
     {
+      corejs: 2,
       loose: true,
       modules: false,
       useBuiltIns: `usage`,

--- a/packages/babel-preset-gatsby/src/index.js
+++ b/packages/babel-preset-gatsby/src/index.js
@@ -45,6 +45,7 @@ module.exports = function preset(_, options = {}) {
       [
         resolve(`@babel/preset-env`),
         {
+          corejs: 2,
           loose: true,
           modules: stage === `test` ? `commonjs` : false,
           useBuiltIns: `usage`,

